lib/newrelic_httprb/instrumentation.rb

```diff
@@ -0,0 +1,33 @@
+DependencyDetection.defer do
+  named :http_rb
+
+  depends_on do
+    defined?(HTTP) && defined?(HTTP::Client)
+  end
+
+  executes do
+    ::NewRelic::Agent.logger.info 'Installing http.rb instrumentation'
+    require 'new_relic/agent/cross_app_tracing'
+    require 'newrelic_http/wrappers'
+  end
+
+  executes do
+    class HTTP::Client
+      def perform_with_newrelic_trace(request, options)
+        wrapped_request = ::NewRelicHTTP::HTTPRequest.new(request)
+
+        ::NewRelic::Agent::CrossAppTracing.tl_trace_http_request(wrapped_request) do
+          # RUBY-1244 Disable further tracing in request to avoid double
+          # counting if connection wasn't started (which calls request again).
+          response = perform_without_newrelic_trace(request, options)
+          ::NewRelicHTTP::HTTPResponse.new(response)
+        end
+
+        response
```

Comment:
Not sure if this will work. Don't you need to initialize `response` before using it in the block? Otherwise the `response` in the block will be local to the block itself.